text/0006-equivocation-catastrophes.md

```diff
@@ -0,0 +1,111 @@
+3. The users agree about the last correct block, the transactions that need to be replayed on top of it and the set of equivocating validators (exchanging evidence if necessary).
+4. A chainspec for the restart is manually constructed and agreed upon. This includes:
+    - the initial set of validators for the restarted network which excludes the equivocators,
+    - the initial global state hash for the restarted network
+5. The network is being restarted using the chainspec constructed above.
+
+## Drawbacks
+
+[drawbacks]: #drawbacks
+
+The single drawback is that this proposal requires manual intervention in the case of an equivocation catastrophe. An ideal situation would be if the network could handle these automatically, but this is most probably impossible.
+
+## Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+As mentioned above, automatic handling of equivocation catastrophes is probably impossible (as it would effectively mean constructing a consensus algorithm that exceeds proven theoretical limitations), so manual intervention is really the single option of dealing with such situations.
+
+An alternative to social consensus would be to have a designated party deal with the situation. However, such a solution would go against the goal of creating a truly distributed system. In this alternative, the party responsible for handling equivocation catastrophes would effectively single-handedly control the network, which is something we want to avoid.
+
+Not handling equivocation catastrophes at all would mean the end of the network once such a situation happens. Thus, even though we expect the probability of an equivocation catastrophe happening to be low (and we are making efforts towards making it as low as possible), the costs in the event of it happening are great enough to justify guarding against it.
+
+## Prior art
+
+[prior-art]: #prior-art
+
+There have been cases of blockchain projects introducing manual interventions in order to revert catastrophical events, for example the Ethereum DAO hack.
+
+## Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+- Which option for choosing the last correct block do we pick?
+- Do we even require the last correct block, or do we just need to agree on an initial global state for the restarted network? Having the starting point for the new network in the form of a block would help in keeping the continuity of the blockchain and can be useful for history archiving purposes, but doesn't seem to be strictly required for keeping the functionality of the network.
+- A related question which will be addressed in a separate CEP is about the process of joining the network by new validators. A new validator will have to be able to join the correct, agreed upon instance of the network, and might have to refer to social consensus in order to get the correct initial data (the correct chainspec).
```

Comment:
That's CEP0005, is it? Would be good to link to it (or the PR) here.

Reply:
Looks like it :+1: